cmd/kubeadm/app/apis/kubeadm/types.go

```diff
@@ -179,6 +157,15 @@ type ImageMeta struct {
 	//TODO: evaluate if we need also a ImageName based on user feedbacks
 }
 
+// AddOn defines settings to be used when deploying a specific addon kind
+type AddOn struct {
+	// Kind defines the addon type
+	Kind string
+
+	// ImageMeta allows to customize the image used for the addon
+	ImageMeta
```

Comment:
I'm not sure assuming all the addons have only one image is correct (e.g. kube-dns has three images)
Additionally, despite asking this several times, it is not yet clear to how we are going to pass this to the addon project, because the assumption to "use kustomize" implies kubeadm should know the internal (e.g. the name of the Deployments/the name of the container) of every possible addon which is not  practically possible

Reply:
That's the reason we introduced ImageMeta as it is in the first place. It allows users to change the image repo and tags, but leave the exact name to kubeadm to manage. In the usual case, all of the images to be used will be on the same repo and even tagged in the same manner. If not, that's easy to fix.
But I see your concern here. We might want to think about this a bit more.